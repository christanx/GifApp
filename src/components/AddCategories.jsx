import { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategories = ({ setCategories }) => {
    const [searchText, setSearchText] = useState('')

    const handleChange = (e) => setSearchText(e.target.value);
    const handleSubmit = (e) => {
        e.preventDefault();
       
        if (searchText.trim().length && searchText !== '') {
            setCategories(categories => [searchText, ...categories]);
            setSearchText('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input id='search' type="text" placeholder='Search' onChange={handleChange} value={searchText} />
            <input id='searchHidden' type="hidden" value={searchText} />
        </form>
    )
}

AddCategories.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategories
