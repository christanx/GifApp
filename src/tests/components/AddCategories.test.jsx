import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import AddCategories from "../../components/AddCategories"


describe('Testing <AddCategories />', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategories setCategories={setCategories}/>);
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategories setCategories={setCategories}/>);
    });

    test('Component successful', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe cambiar la caja de texto "input"', () => {
        const input = wrapper.find('#search');
        const value = "Hello world";
        input.simulate('change', { target: { value } });
        const textInput = wrapper.find('#searchHidden').prop('value').trim();
        expect(textInput).toBe(value);
    })
    
    // test('No debe enviar ninguna informaciÃ³n con submit', () => {
    //     wrapper.find('form').simulate('submit', { preventDefault: () => {} })
    //     expect(setCategories).not.toHaveBeenCalled();
    // })

    test('Debe ejecutar el setCategories y limpiar la caja de texto', () => {
        const input = wrapper.find('#search');
        const value = "Hello christian !!!ðŸ˜‚";
        input.simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(wrapper.find('#search').prop('value')).toBe('');
    })
    
    
})
