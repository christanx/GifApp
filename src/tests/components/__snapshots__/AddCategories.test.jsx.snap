// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AddCategories /> Component successful 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <input
    id="searchHidden"
    type="hidden"
    value=""
  />
</form>
`;
